### PR TITLE
Standardizing links to hydroshare and ways of creating them.

### PR DESCRIPTION
**Pull request by alexlemann**

### CODE REVIEW

**Comment by pkdash** [comment]
@alexlemann I will try to review this today. In the mean time if you can resolve the file conflicts that would be great.

**Comment by alexlemann** [comment]
@pkdash Okay. I'll review it today too :). It looks a bit rough on review. I'm definitely interested in lots of feedback including on the overall idea of it.

File: hs_core/views/__init__.py [hunk 1]
```diff
@@ -1344,7 +1345,7 @@ def get_user_or_group_data(request, user_or_group_id, is_group, *args, **kwargs)
 
         user_data['name'] = user_name
         user_data['email'] = user.email
-        user_data['url'] = '{domain}/user/{uid}/'.format(domain=utils.current_site_url(), uid=user.pk)
+        user_data['url'] = '{domain}/user/{uid}/'.format(domain=current_site_url(), uid=user.pk)
```
Comment (pkdash) [inline hs_core/views/__init__.py]: Should this be:

```user_data['url'] = current_site_url('/user/{uid}'.format(uid=user.pk))```

File: hs_collection_resource/migrations/0003_collectiondeletedresource_resource_url.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+from django.db import migrations, models
+
+from hs_core.hydroshare.utils import current_site_url
+
+
+def initial_value(apps, schema_editor):
+    CollectionDeletedResource = apps.get_model('hs_collection_resource', 'CollectionDeletedResource')
+    for o in CollectionDeletedResource.objects.all():
+        o.resource_url = current_site_url('/resource/{}'.format(o.resource_id))
```
Comment (pkdash) [inline hs_collection_resource/migrations/0003_collectiondeletedresource_resource_url.py]: I am trying to find out where in code this new attribute is being set.

**Comment by pkdash** [comment]
@alexlemann I like you approach of using the utility function ```current_site_url()``` for all url generations.

**Comment by pkdash** [comment]
@alexlemann I have left couple of comments. There seems to be file conflicts that you need to resolve.

**Comment by pkdash** [comment]
@alvacouch Can you review this too?

**Comment by dtarb** [comment]
@alvacouch please look at this and decide if this PR is needed or not.  If not, lets close this.

**Pull request closed without merging**